### PR TITLE
Decoupling: Move WordPress media integration to `wp-story-editor`

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

Move `wp.media` related code to wp-story-editor

## To-do

- [x] Create `MediaUpload` component.
- [x] Move `useMediaPicker` to `wp-story-editor`.
- [x] Use `MediaUpload` in core editor.
- [x] Update all jest tests according to the new code change. ( Provide `MediaUpload` via config )
- [x] Move `getResourceFromAttachment` to `wp-story-editor`.
- [x] Move `getResourceFromMediaPicker` to `wp-story-editor`
- [x] Update relevant object structures after moving `getResourceFromMediaPicker` .
- [x] Update integration test.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses https://github.com/google/web-stories-wp/issues/2918
See: https://github.com/google/web-stories-wp/issues/8953

### CODE REVIEW

**Comment by swissspidy** [comment]
FYI, #8990 caused a few merge conflicts here

File: packages/story-editor/src/components/panels/design/captions/captions.js [hunk 1]
```diff
@@ -127,11 +126,11 @@ function CaptionsPanel({ selectedElements, pushUpdate }) {
...
   const handleChangeTrack = useCallback(
-    (attachment) => {
+    ({ src = '', id }) => {
       const newTracks = {
-        track: attachment?.url,
-        trackId: attachment?.id,
-        trackName: attachment?.filename,
+        track: src,
+        trackId: id,
+        trackName: src.split('/').pop(),
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/design/captions/captions.js:133]: Could use use [getFileNameFromUrl](https://github.com/google/web-stories-wp/blob/main/packages/media/src/getFileNameFromUrl.js) here?

Comment (mohdsayed) [reply packages/story-editor/src/components/panels/design/captions/captions.js:133]: The existing code `attachment?.filename` gives the file name with extension ( for example caption.vtt ) however, [getFileNameFromUrl](https://github.com/google/web-stories-wp/blob/main/packages/media/src/getFileNameFromUrl.js) will return only the file name without extension.

Comment (swissspidy) [reply packages/story-editor/src/components/panels/design/captions/captions.js:133]: Don‘t think we need to store trackName here in the first place.

**Comment by mohdsayed** [comment]
The build had gone crazy, finally passed.

@swissspidy Tests fixed now!

**Comment by swissspidy** [comment]
@sayedtaqui the e2e tests still seem to be failing

**Comment by mohdsayed** [comment]
@swissspidy This one "E2E Tests / chrome - WP 5.5 (pull_request)" ?

**Comment by swissspidy** [comment]
> @swissspidy This one "E2E Tests / chrome - WP 5.5 (pull_request)" ?

Yes, but also "E2E Tests / chrome - WP latest (pull_request)".

We test against WP latest & WP 5.5. Both e2e tests are failing currently in this PR.

They pass on `main`.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**